cpu-features/src/cpu-features.c

```diff
@@ -186,15 +218,37 @@ hacl_init_cpu_features()
   _sse42 = (ecx & ECX_SSE4_2) != 0;
 #endif
 
+#if defined(CPU_FEATURES_LINUX) && defined(CPU_FEATURES_ARM64) &&    \
```

Comment:
Unfortunately, I don't have arm64 machine with Windows on it neither have access to such instance. I'd like to know any possible options to achieve that.

Reply:
I believe this the default execution flow already, `hacl_init_cpu_features()` doesn't do any configuration on Windows which means the global variables are left with the initial value `0`. It flows down ultimately to the portable implementation.

At first, I thought you would mean to implement feature detection on Windows ARM. It makes sense to have it works at least for x86 which is more important. I will see what I can do here.

Reply:
`__cpuid` function is a good choice to avoid having assembly file that utilizes `cpuid` instruction since inline assembly isn't supported for MSVC x64. We still need to have inline assembly of cpuid for other compilers on Windows, fortunately we can use `cpuid()` that already exist for Linux but I don't understand why this function preserves `ebx` register for 32-bit version https://github.com/cryspen/hacl-packages/blob/main/cpu-features/src/cpu-features.c#L65 and not do that for x64 version. I disassembled x64 version of the function built with clang and the generated machine code does preserve `rbx` register since it's specified in the input operands https://github.com/cryspen/hacl-packages/blob/main/cpu-features/src/cpu-features.c#L51
Technically, both versions are supposed to work properly I just think it's good to have identical scheme for both if possible. 32-bit scheme seems more reasonable to me because I'm not sure why the compiler would preserve input operand of inline assembly.

Reply:
I made this change `#elif defined(__GNUC__)` to `#elif defined(__linux__)` https://github.com/mamonet/hacl-packages/commit/4b5f3eda6c7236bd33bf819db805547fd80b4487#diff-fff0c10580c2c65492a930241a62b51ac2cc56c8f0ff4ea59f5a42e38d43bd62L30
Let me know if it implies other conditions I didn't cover

Reply:
I think this should work.